essentials/src/api/subxt_wrapper.rs

```diff
@@ -690,16 +671,28 @@ async fn subxt_unpin_chain_head(api: &OnlineClient<PolkadotConfig>, sub_id: &str
 	Ok(Response::UnpinChainHead(api.rpc().chainhead_unstable_unpin(sub_id.to_string(), hash).await?))
 }
 
-fn subxt_extract_parainherent(block: &subxt::rpc::types::ChainBlock<PolkadotConfig>) -> Result {
-	// `ParaInherent` data is always at index #1.
-	let bytes = &block.extrinsics[1].0;
-
-	let data = match decode_extrinsic(&mut bytes.as_slice()).expect("Failed to decode `ParaInherent`") {
-		SubxtCall::ParaInherent(
+async fn subxt_extract_parainherent(
+	block: &subxt::blocks::Block<PolkadotConfig, OnlineClient<PolkadotConfig>>,
+) -> Result {
+	let body = block.body().await?;
+	let ex = body
+		.extrinsics()
+		.iter()
+		.take(2)
+		.last()
+		.expect("`ParaInherent` data is always at index #1")
+		.expect("`ParaInherent` data must exist");
+
+	let data = match ex
+		.as_root_extrinsic::<polkadot::Call>()
```

Comment:
nit: This works as intended, we also have the ability to decode directly to a pallet type

```
let data =  match ex.as_extrinsic<polkadot::para_inherent::calls::types::Enter>() 
```

We have a testing example [here](https://github.com/paritytech/subxt/blob/fd853b9b72b5335948d0a54c33cf3a05062caee4/testing/integration-tests/src/full_client/blocks/mod.rs#L170-L173), but again totally up to you since it works